src/chain/mod.rs

```diff
@@ -236,13 +242,36 @@ impl ChainSource {
 
 	pub(crate) fn start(&self, runtime: Arc<tokio::runtime::Runtime>) -> Result<(), Error> {
 		match self {
-			Self::Electrum { server_url, electrum_runtime_client, logger, .. } => {
+			Self::Electrum {
+				server_url,
+				electrum_runtime_client,
+				registered_txs_cache,
+				registered_outputs_cache,
+				logger,
+				..
+			} => {
 				let mut locked_client = electrum_runtime_client.write().unwrap();
-				*locked_client = Some(ElectrumRuntimeClient::new(
-					server_url.clone(),
-					runtime,
-					Arc::clone(&logger),
-				)?);
+				let client =
+					ElectrumRuntimeClient::new(server_url.clone(), runtime, Arc::clone(&logger))?;
+
+				// Apply any previously-cached `Filter` entries
+				{
+					let registered_txs_cache =
+						core::mem::take(&mut *registered_txs_cache.lock().unwrap());
+					for (txid, script_pubkey) in registered_txs_cache {
+						client.register_tx(&txid, &script_pubkey);
+					}
+				}
+
+				{
+					let registered_outputs_cache =
+						core::mem::take(&mut *registered_outputs_cache.lock().unwrap());
+					for output in registered_outputs_cache {
+						client.register_output(output)
+					}
+				}
```

Comment:
The `registered_txs_cache` and `registered_outputs_cache` are in-memory types. This means that every time we stop and then restart the `Node`, these fields will be empty. Doesn't this mean that any previously cached entries would be lost when we stop, and upon restart, we'd have no transactions or outputs to iterate through and then register?